src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangas.kt

```diff
@@ -0,0 +1,208 @@
+        val pagesData = json.decodeFromString<PageDataDto>(decodedData)
+        return pagesData.data.getImages(langOption.pageDelimiter).mapIndexed { index, imageUrl ->
+            Page(index, imageUrl = imageUrl)
+        }
+    }
+
+    private fun findChapterPassword(document: Document): String {
+        return try {
+            val regxPaswdUrl = """\/pages\/%5Btype%5D\/%5Bidmanga%5D\/%5Biddetail%5D-.+\.js""".toRegex()
+            val regxFindPaswd = """AES\.decrypt\(\w+,"(?<password>[^"]+)"\)""".toRegex(RegexOption.MULTILINE)
+            val jsDecryptUrl = document.select("script")
+                .map { it.absUrl("src") }
+                .first { regxPaswdUrl.find(it) != null }
+            val jsDecrypt = client.newCall(GET(jsDecryptUrl, headers)).execute().asJsoup().html()
+            regxFindPaswd.find(jsDecrypt)?.groups?.get("password")!!.value.trim()
+        } catch (e: RuntimeException) {
+            throw RuntimeException("Password not found")
+        }
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val dto = response.asJsoup().htmlDocumentToDto()
+        return MangasPage(
+            mangas = dto.toPopularSManga(),
+            hasNextPage = dto.hasNextPageToPopularMangas(),
+        )
+    }
+
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/${langOption.infix}")
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val maxResult = 6
+        val url = "$apiUrl/api/$langApiInfix/searchforms/$maxResult/".toHttpUrl().newBuilder()
+            .addPathSegment(query)
+            .addPathSegment("${page - 1}")
+            .build()
+        return GET(url, apiHeaders(url.toString()))
+    }
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        if (query.startsWith(slugPrefix)) {
+            val mangaUrl = query.substringAfter(slugPrefix)
+            return client.newCall(GET("$baseUrl/${langOption.infix}/$mangaUrl", headers))
+                .asObservableSuccess().map { response ->
+                    val manga = mangaDetailsParse(response).apply {
+                        url = mangaUrl
+                    }
+                    MangasPage(listOf(manga), false)
+                }
+        }
+        return super.fetchSearchManga(page, query, filters)
+    }
+
+    override fun imageUrlParse(response: Response): String = ""
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val mangasDto = response.parseAs<PageableMangaListDto>()
+        return MangasPage(
+            mangas = mangasDto.toSManga(langOption.infix),
+            hasNextPage = mangasDto.hasNextPage(),
+        )
+    }
+
+    private fun Document.htmlDocumentToDto(): UnionMangasDto {
+        val jsonContent = selectFirst("script#__NEXT_DATA__")!!.html()
+        return json.decodeFromString<UnionMangasDto>(jsonContent)
+    }
```

Comment:
```suggestion
    private inline fun <reified T> Response.parseNextData(): T {
        val jsonContent = asJsoup().selectFirst("script#__NEXT_DATA__")!!.html()
        return json.decodeFromString<NextData<T>>(jsonContent)
    }
```

use it as: `response.parseNextData<List<PopularMangaDto>>()` etc